File: primes/primes.c
#include <stdio.h>

int main(void){
    long long limit = 1000000;
    long long count = 0;
    long long current = 2;
    while (current <= limit) {
        int is_prime = 1;
        if (current != 2 && current % 2 == 0){
            is_prime = 0;
        }
          
        long long cand = 3;
        while (is_prime == 1 && cand * cand <= current){
            if (current % cand == 0)
              is_prime = 0;
            cand++;
        }
        if (is_prime == 1)
            ++count;
        ++current;
    }
    printf("Total = %lld\n", count);
}

File: primes/primes.py
limit = 1000000
count = 0
current = 2
while current <= limit:
    is_prime = 1
    if current != 2 and current % 2 == 0:
        is_prime = 0
    candidate = 3
    while is_prime == 1 and candidate * candidate <= current:
        if current % candidate == 0:
            is_prime = 0
        candidate += 1
    if is_prime == 1:
        #print(current)
        count += 1
    current += 1
print("Total =", count)
